app/MMB/src/scenes/timeline/Timeline.js

```diff
@@ -53,35 +54,53 @@ function getValidTimelineEvent(data) {
   return null;
 }
 
-function renderDaySeparator(event, prevEvent) {
-  if (event && event.timestamp && prevEvent && prevEvent.timestamp) {
-    const eventTimestamp = new Date(event.timestamp);
-    const prevEventTimestamp = new Date(prevEvent.timestamp);
-    if (!DateUtils.isSameDay(eventTimestamp, prevEventTimestamp)) {
-      return <DaySeparator date={eventTimestamp} />;
+function renderDaySeparator(renderedEvent, prevRenderedEvent) {
+  const timestamp = idx(renderedEvent, _ => _.props.data.timestamp);
+  if (!prevRenderedEvent) {
+    if (timestamp) {
+      return <DaySeparator date={new Date(timestamp)} />;
+    }
+  } else {
+    const prevTimestamp = idx(prevRenderedEvent, _ => _.props.data.timestamp);
+    if (timestamp && prevTimestamp) {
+      const eventTimestamp = new Date(timestamp);
+      const prevEventTimestamp = new Date(prevTimestamp);
+      if (!DateUtils.isSameDay(eventTimestamp, prevEventTimestamp)) {
+        return <DaySeparator date={eventTimestamp} />;
+      }
     }
   }
   return null;
```

Comment:
Could you please try to refactor this function a bit? The level of nested if/else makes it hard to read and understand.

Reply:
Ok, will do.

Reply:
Done